modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/TableCell.js

```diff
@@ -75,48 +75,48 @@ function TableCell({
 	valuePath,
 	view,
 }) {
-	let dataRenderer = DefaultRenderer;
-
 	const {customDataRenderers, inlineEditingSettings} = useContext(
 		DataSetContext
 	);
 
+	const [loading, setLoading] = useState(false);
+
 	const contentRenderer = view.contentRenderer;
 
+	let SyncDataRenderer = DefaultRenderer;
+
 	if (contentRenderer) {
 		if (customDataRenderers && customDataRenderers[contentRenderer]) {
-			dataRenderer = customDataRenderers[contentRenderer];
+			SyncDataRenderer = customDataRenderers[contentRenderer];
 		}
 		else {
-			dataRenderer = getDataRendererById(view.contentRenderer);
+			SyncDataRenderer = getDataRendererById(contentRenderer);
 		}
 	}
 
 	if (view.contentRendererModuleURL) {
-		dataRenderer = null;
+		SyncDataRenderer = null;
 	}
 
-	const [currentView, setCurrentView] = useState({
-		...view,
-		Component: dataRenderer,
-	});
-
-	const [loading, setLoading] = useState(false);
+	const [DataRenderer, setDataRenderer] = useState(() => SyncDataRenderer);
 
 	useEffect(() => {
-		if (!loading && currentView.contentRendererModuleURL) {
+		if (!loading && view.contentRendererModuleURL && !DataRenderer) {
 			setLoading(true);
-			getDataRendererByURL(currentView.contentRendererModuleURL).then(
-				(Component) => {
-					setCurrentView({
```

Comment:
Hi @markocikos, this PR fixes an infinite loop caused by this `useEffect`. 
It happened when a FDS required a data renderer that came from a different module and was caused by this `setCurrentView` within a `useEffect` with `currentView` in the dependency array.

I bet it's me who wrote it one or two years ago! :(